src/backend/optimizer/plan/createplan.c

```diff
@@ -6343,29 +6507,33 @@ prepare_sort_from_pathkeys(Plan *lefttree, List *pathkeys,
  *	  'relids' is the set of relations required by prepare_sort_from_pathkeys()
  */
 static Sort *
-make_sort_from_pathkeys(Plan *lefttree, List *pathkeys, Relids relids)
+make_sort_from_pathkeys(PlannerInfo *root, Plan *lefttree,
+						List *pathkeys, Relids relids)
 {
 	int			numsortkeys;
 	AttrNumber *sortColIdx;
 	Oid		   *sortOperators;
 	Oid		   *collations;
 	bool	   *nullsFirst;
+	bool        mkqsApplicable;
 
 	/* Compute sort column info, and adjust lefttree as needed */
-	lefttree = prepare_sort_from_pathkeys(lefttree, pathkeys,
+	lefttree = prepare_sort_from_pathkeys(root,
+										  lefttree, pathkeys,
 										  relids,
 										  NULL,
 										  false,
 										  &numsortkeys,
 										  &sortColIdx,
 										  &sortOperators,
 										  &collations,
-										  &nullsFirst);
+										  &nullsFirst,
+										  &mkqsApplicable);
 
 	/* Now build the Sort node */
 	return make_sort(lefttree, numsortkeys,
 					 sortColIdx, sortOperators,
-					 collations, nullsFirst);
```

Comment:
Seems like we only set `mkqsApplicable` here in PR, other places set `NULL`, Do we support mksort for other scenarios like MergeAppend, GroupAgg, or only support Sort explicitly.

Reply:
To be precise, for now mksort supports only the code path with `make_sort_from_pathkeys()`. Theoretically it can support others, but I prefer to complete the code after we have a decision about the `mkqsApplicable` related code in optimizer.